utils/_booking.py

```diff
@@ -239,20 +189,16 @@ class Count(Action):
 
 class Histogram(Action):
     def __init__(
-            self, variable,
-            edges = None, name = None):
-        Action.__init__(self, variable, name)
-        if isinstance(variable, Binning) and edges is None:
-            self.binning = variable
-        else:
-            self.binning = Binning(
-                variable, edges)
+            self, name,
+            variable, edges):
+        Action.__init__(self, name, variable)
+        self.edges = edges
+        self.nbins = len(edges) - 1
```

Comment:
same as above, you do not need that member since `edges` is there. you just have to keep it in sync with `edges`. but it's just a detail. keep it or remove it, as you like. if you use `nbins` like only once, i would put the impl of `nbins` directly there.